code/modules/admin/verbs/modifyvariables.dm

```diff
@@ -94,401 +206,205 @@ var/list/VVckey_edit = list("key", "ckey")
+			message_admins("[key_name_admin(src)] modified [original_name]'s list [objectvar]: CLEAR DUPES")
+			return
 
-	else if(istext(variable))
-		usr << "Variable appears to be <b>TEXT</b>."
-		default = "text"
+		if(variable == "(SHUFFLE)")
+			L = shuffle(L)
+			if (!O.vv_edit_var(objectvar, L))
+				src << "Your edit was rejected by the object."
+				return
+			world.log << "### ListVarEdit by [src]: [O.type] [objectvar]: SHUFFLE"
+			log_admin("[key_name(src)] modified [original_name]'s [objectvar]: SHUFFLE")
+			message_admins("[key_name_admin(src)] modified [original_name]'s list [objectvar]: SHUFFLE")
+			return
 
-	else if(isloc(variable))
-		usr << "Variable appears to be <b>REFERENCE</b>."
-		default = "reference"
+		index = names[variable]
 
-	else if(isicon(variable))
-		usr << "Variable appears to be <b>ICON</b>."
-		variable = "\icon[variable]"
-		default = "icon"
 
-	else if(istype(variable,/client))
-		usr << "Variable appears to be <b>CLIENT</b>."
-		default = "cancel"
+	var/assoc_key
+	if (index == null)
+		return
+	var/assoc = 0
+	var/prompt = alert(src, "Do you want to edit the key or it's assigned value?", "Associated List", "Key", "Assigned Value", "Cancel")
+	if (prompt == "Cancel")
+		return
+	if (prompt == "Assigned Value")
+		assoc = 1
+		assoc_key = L[index]
+	var/default
+	var/variable
+	if (assoc)
+		variable = L[assoc_key]
+	else
+		variable = L[index]
 
-	else if(istype(variable,/atom) || istype(variable,/datum))
-		usr << "Variable appears to be <b>TYPE</b>."
-		default = "type"
+	default = vv_get_class(variable)
 
-	else if(istype(variable,/list))
-		usr << "Variable appears to be <b>LIST</b>."
-		default = "list"
+	src << "Variable appears to be <b>[uppertext(default)]</b>."
 
-	else
-		usr << "Variable appears to be <b>FILE</b>."
-		default = "file"
-
-	usr << "Variable contains: [variable]"
-	if(dir)
-		switch(variable)
-			if(1)
-				setDir("NORTH")
-			if(2)
-				setDir("SOUTH")
-			if(4)
-				setDir("EAST")
-			if(8)
-				setDir("WEST")
-			if(5)
-				setDir("NORTHEAST")
-			if(6)
-				setDir("SOUTHEAST")
-			if(9)
-				setDir("NORTHWEST")
-			if(10)
-				setDir("SOUTHWEST")
-			else
-				setDir(null)
-
-		if(dir)
-			usr << "If a direction, direction is: [dir]"
-
-	var/class = "text"
-	if(src.holder && src.holder.marked_datum)
-		class = input("What kind of variable?","Variable Type",default) as null|anything in list("text",
-			"num","type","reference","mob reference", "icon","file","list","edit referenced object","restore to default", "new atom", "new datum","marked datum ([holder.marked_datum.type])", "DELETE FROM LIST")
-	else
-		class = input("What kind of variable?","Variable Type",default) as null|anything in list("text",
-			"num","type","reference","mob reference", "icon","file","list","edit referenced object","restore to default", "new atom", "new datum", "DELETE FROM LIST")
+	src << "Variable contains: [L[index]]"
 
-	if(!class)
-		return
+	if(default == VV_NUM)
+		var/dir_text = ""

```

Comment:
could probably do with a proc for this.
dir2dirtext or something (it might actually exist already)